### DISCUSSION TITLE
Крашится загрузка карты

### DISCUSSION THREAD

**Discussion by vermorag** [yandex-maps-unofficial/vue-yandex-maps#296] [Q&A (2.x)] [answered]
[`/packages/vue-yandex-maps/src/functions/init.ts:66`](https://github.com/yandex-maps-unofficial/vue-yandex-maps/blob/master/packages/vue-yandex-maps/src/functions/init.ts#L66)
Тут точно должно быть просто сравнение?
На [`/packages/vue-yandex-maps/src/functions/init.ts:111`](https://github.com/yandex-maps-unofficial/vue-yandex-maps/blob/master/packages/vue-yandex-maps/src/functions/init.ts#L111) оно в дефолт встает в {}, а оно всегда валидируется в true.
![image](https://github.com/yandex-maps-unofficial/vue-yandex-maps/assets/25276845/e4faa633-05aa-4f04-83fc-5913430bb5f0)
 
Я понимаю, что это верхушка айсберга, но конректно сейчас не грузится карта и все падает на 68 строчке, где он пытается `getDefaultConfig()` вызвать, а в это место приходит через if.

В консоли с крашем карты сейчас: 
![image](https://github.com/yandex-maps-unofficial/vue-yandex-maps/assets/25276845/9cb4a941-d3ed-4ac9-a314-90374dd04e32)
При этом ymaps3 там существует, но дефолтного конфига там нет
![image](https://github.com/yandex-maps-unofficial/vue-yandex-maps/assets/25276845/dc1c6a26-7f6a-49c7-8e62-80c030c6abb7)

**Comment by daniluk4000** [comment]
Такс. Я согласен, что можно это лишний раз не дергать, но вопрос, почему defaultConfig undefined после ready... 

Это всегда воспроизводится?

**Reply by vermorag** [reply]
В этом и прикол, что только у одного клиента. Ключ Я.Карты действительный, пробовал разные подставить - результат один.
При этом сам скрипт с серверов яндекса он стабильно подгружает. Была бы проблема в ключе - он бы 403 отдавал.

**Reply by daniluk4000** [reply, marked as answer]
Верно.

Я сталкивался с тем, что ready на самом деле не сигнализирует о том, что карта точно готова к работе, и иногда мы ловим странные ошибки. При этом стабильно я это никогда не мог воспроизвести.

Попробую поиграться со скоростью интернета / троттлингом. Но имхо - выглядит как баг на стороне яндекса (хоть и может починиться на нашей стороне).

А вызов сета ключей при пустоте я починю) Но тут скорее минорное улучшение скорости запуска, а не фикс бага

**Reply by vermorag** [reply]
Спасибо. Не спорю, поэтому и не стал заводить именно баг, а пошел в обсуждения. Просто будет минус одно предупреждение об ошибке, вдруг другие появятся.

**Reply by vermorag** [reply]
Поправил в сборке руками. И сейчас получаю гениальную ошибку от яндекса:
`Error: ymaps3.import: no loader for pkg @yandex/ymaps3-clusterer@0.0.1`

**Reply by daniluk4000** [reply]
Ну вот и ответ))) 

А там других ошибок нет, да, я так понимаю?

И еще: попробуй по приколу в сборке слип(1000) бахнуть. Если починится, то моя теория верна

**Reply by vermorag** [reply]
Она на один раз появилась, потом пропала. Сейчас попробую впихнуть
![image](https://github.com/yandex-maps-unofficial/vue-yandex-maps/assets/25276845/f71cb271-2e0a-470c-80e7-0b2ba212f7c4)

**Reply by daniluk4000** [reply]
![image](https://github.com/yandex-maps-unofficial/vue-yandex-maps/assets/10450717/b2eb65f7-7038-4e11-8db7-ab0946905f88)

У нас вот такая штука появляется без причины в минимальном количестве

**Reply by daniluk4000** [reply]
Кстати)

![image](https://github.com/yandex-maps-unofficial/vue-yandex-maps/assets/10450717/c7a6a344-29d2-443b-a620-102ac965cec9)

**Reply by vermorag** [reply]
Воткнул слип на 1 секунду. Сейчас как раз `TypeError: ymaps3.YMapDefaultSchemeLayer is not a constructor` и `TypeError: ymaps3.YMapDefaultFeaturesLayer is not a constructor`.

Не понятно, что у клиента это воспроизводится на 100% и всегда, а одновременно с этим на стенде с таким же кодом не воспроизводится.

Ради эксперимента проверил - и там, и там 3.0.13863142 версия

**Reply by daniluk4000** [reply]
Ерунда какая-то. Я бы писал Яндексу в форму ОС со всеми этими ошибками. Если слип не помог, то что-то жестко фейлится при загрузке карты... 

Можно еще проверить, баг ли это в либе -- создать очень базовую карту напрямую. Но вроде как либа подключает всё по правилам

**Reply by daniluk4000** [reply]
В доке яндекса еще нет defer/async/type. Можно их убрать, но тоже не должно никак влиять совершенно

**Reply by vermorag** [reply]
Видимо приедтся им писать. Но вот предсказываю ответ будет таким: 
![image](https://github.com/yandex-maps-unofficial/vue-yandex-maps/assets/25276845/8ca8a099-9d82-43d2-a1b7-a9f8fb1ed50f)

**Reply by daniluk4000** [reply]
Я причем недавно проверял, так и не добавили type button)

**Reply by vermorag** [reply]
![image](https://github.com/yandex-maps-unofficial/vue-yandex-maps/assets/25276845/eae60c5c-6e4b-453b-83a1-a36a3dc64729)
Видимо, без них точно не разобраться

**Reply by daniluk4000** [reply]
Ну да, что-то совсем жесть. Если оно даже нативно так

**Reply by vermorag** [reply]
Чисто ради эксперимента, можешь глянуть что в мете грузиться у первого скрипта? Там начало вот такое.
![image](https://github.com/yandex-maps-unofficial/vue-yandex-maps/assets/25276845/45eebd39-4a22-41a4-be03-e5dd281f458c)

Мне интересно что версиях слоев и фичах
![image](https://github.com/yandex-maps-unofficial/vue-yandex-maps/assets/25276845/0dd99588-b14a-4a4d-ae6d-6f90609e534d)
![image](https://github.com/yandex-maps-unofficial/vue-yandex-maps/assets/25276845/37461f4f-498d-4c2b-8b0e-ff34af25ce55)


Сравниваю версии со стендом своим, отличие только у версии trf (что бы это не значило), вот интересно -оно у всех разное?
На рабочем стенде оно 1713451200, на нерабочем 1713452220.

Может какое канареечное разветывание у них так?

**Reply by daniluk4000** [reply]
```json
 {
            "meta": {
                "lang": "ru_RU",
                "hosts": {
                    "apiCoverageService": "https://api-maps.yandex.ru/services/coverage/",
                    "apiRouteService": "https://api-maps.yandex.ru/services/route/",
                    "apiSearchService": "https://api-maps.yandex.ru/services/search/",
                    "apiConfigService": "https://api-maps.yandex.ru/v3/config",
                    "mapTiles": "https://core-renderer-tiles.maps.yandex.net/tiles?l=map&%c&%l",
                    "mapjTiles": "https://core-renderer-tiles.maps.yandex.net/tiles?l=mapj&%c&%l&experimental_disable_toponym_hotspots=true",
                    "satTiles": "https://core-sat.maps.yandex.net/tiles?l=sat&%c&%l",
                    "vectorTiles": "https://core-renderer-tiles.maps.yandex.net/vmap2/tiles?lang={{lang}}&x={{x}}&y={{y}}&z={{z}}&zmin={{zmin}}&zmax={{zmax}}&v={{version}}",
                    "vectorImages": "https://core-renderer-tiles.maps.yandex.net/vmap2/icons?id={{id}}&scale={{scale}}",
                    "vectorMeshes": "https://core-renderer-tiles.maps.yandex.net/vmap2/meshes?id={{id}}",
                    "vectorGlyphs": "https://core-renderer-tiles.maps.yandex.net/vmap2/glyphs?lang={{lang}}&font_id={{fontId}}&range={{range}}",
                    "suggestApi": "https://suggest-maps.yandex.ru/suggest-geo",
                    "routerApi": "https://api.routing.yandex.net/v2/route",
                    "searchApi": "https://search-maps.yandex.ru/"
                },
                "token": "",
                "layers": {
                    "map": {
                        "version": "24.04.18-0-b240404173730",
                        "scaled": true,
                        "hotspotZoomRange": [1, 23]
                    },
                    "skl": {
                        "version": "24.04.18-0-b240404173730",
                        "scaled": true,
                        "hotspotZoomRange": [1, 23]
                    },
                    "sta": {
                        "version": "2024.04.17.19.12-1_24.04.16-0-19425",
                        "scaled": false
                    },
                    "stv": {
                        "version": "2024.04.17.19.12-1_24.04.16-0-19425",
                        "scaled": false
                    },
                    "sat": {
                        "version": "3.1201.0",
                        "scaled": false
                    },
                    "trf": {
                        "version": "1713452640",
                        "scaled": true
                    },
                    "trfe": {
                        "version": "",
                        "scaled": true,
                        "hotspotZoomRange": [0, 21]
                    }
                },
                "allowedFeatures": {
                    "vector": true,
                    "customization": true,
                    "customLayers": false,
                    "router": false,
                    "experiments": {},
                    "deprecatedSuggest": true,
                    "deprecatedRouter": true,
                    "deprecatedSearch": true,
                    "tile3d": false
                },
                "geolocation": {
                    "lat": 55.753215,
                    "long": 37.622504,
                    "latSpan": 0.878654,
                    "longSpan": 1.164423
                },
                "metrics": {
                    "allowSending": true,
                    "counterId": "80578111",
                    "pageUrl": "api-maps.yandex.ru/v3"
                },
                "copyrights": {
                    "userAgreementHost": "https://yandex.ru/legal/maps_termsofuse/?lang={{lang}}",
                    "userAgreementTextLong": "Условия использования",
                    "userAgreementTextShort": "Условия",
                    "logoLang": "ru",
                    "allowRemove": false
                },
                "ruler": {
                    "kmText": "км",
                    "mText": "м",
                    "mileText": "ми"
                },
                "apikey": "" 
            },
            "src": {
                "base": "https://yastatic.net/s3/front-maps-static/maps-front-jsapi-3/3.0.13863142/build/static/bundles"
            }
        };
```

**Reply by daniluk4000** [reply]
Блин может у него кэш какой то? В приватной вкладке тоже не работает?

**Reply by vermorag** [reply]
Там у нескольких человек за раз не работает. Однозначно не одиночный кэш.

**Reply by daniluk4000** [reply]
Потом окажется, что Яндекс в аб тесте тестируют что нибудь)

**Reply by vermorag** [reply]
Вот специально взял один ключ и 2 домена. Разница только тут: 
![image](https://github.com/yandex-maps-unofficial/vue-yandex-maps/assets/25276845/0e92d5fa-e5bb-447c-b1e1-abec38b12fa3)
А с тобой версии карт тоже разные

**Reply by vermorag** [reply]
Слушай. А в твоем плагине есть проверка на наличие уже готового ymaps3? Чтобы его лишний раз не загружать

**Reply by daniluk4000** [reply]
Ага, init.ts, строка 21-22

**Reply by daniluk4000** [reply]
Кстати говоря, нет проверки на ready в этом коде, так что советую подождать ready перед инитом карты

Проверку я добавлю, чет упущение

**Reply by vermorag** [reply]
Сейчас ради эксперимента взял код инициализации с рабочего стенда, перенес на не рабочий. И прям в консоли ловлю
![image](https://github.com/yandex-maps-unofficial/vue-yandex-maps/assets/25276845/e02fc84b-fb50-47de-bf74-4b6b1a2b702d)
Это без запроса к яндексу

self.__chunk_yandex_ymaps3 = self.__chunk_yandex_ymaps3

Еще есть self.location.href

Посмотрел на рабочем стенде. self должен быть = window

**Reply by daniluk4000** [reply]
А что это за код? Это их скрипт?

**Reply by daniluk4000** [reply]
Ага, ну да, их скрипт

**Reply by vermorag** [reply]
Они вместо window используют self. А он на том стенед почему-то пустая строка

**Reply by daniluk4000** [reply]
Выглядит как костыли сборки пятого вебпака

**Reply by vermorag** [reply]
Нашел где self="", выпилил принудительно руками. Все заработало.

Огромное спасибо за помощь. Тогда по дискуссии остается только предложение добавить проверку, чтобы лишний код не дергать.

Еще в свою карту докинул проверку, что self - это именно window, а то мало ли...

**Reply by daniluk4000** [reply]
Да, проверку для конфига добавлю. Всегда пожалуйста!

Ну и дичь, конечно)
